frontend/packages/core/src/Input/text-field.tsx

```diff
@@ -44,39 +45,42 @@ const StyledAutocomplete = styled(Autocomplete)({
   },
 });
 
-const StyledTextField = styled(BaseTextField)({
-  "--error-color": "#DB3615",
-  height: "unset",
+const TEXT_FIELD_COLOR_MAP = {
+  default: "rgba(13, 16, 48, 0.6)",
+  inputDefault: "rgba(13, 16, 48, 0.38)",
+  error: "#DB3615",
+  warning: "#FCD34D",
+  focused: "#3548d4",
+};
 
+const StyledTextField = styled(BaseTextField)<{
+  $color?: MuiStandardTextFieldProps["color"];
+}>({}, props => ({
+  height: "unset",
   ".MuiInputLabel-root": {
-    "--label-color": "rgba(13, 16, 48, 0.6)",
-
-    color: "var(--label-color)",
-
+    color: `${TEXT_FIELD_COLOR_MAP[props.color] || TEXT_FIELD_COLOR_MAP.default}`,
     "&.Mui-focused": {
-      color: "var(--label-color)",
+      color: `${TEXT_FIELD_COLOR_MAP[props.color] || TEXT_FIELD_COLOR_MAP.default}`,
     },
     "&.Mui-error": {
-      color: "var(--error-color)",
+      color: `${TEXT_FIELD_COLOR_MAP.error}`,
     },
   },
-
   ".MuiInputBase-root": {
     "--input-border-width": "1px",
-    "--input-default-color": "rgba(13, 16, 48, 0.38)",
     borderRadius: "4px",
     fontSize: "16px",
     backgroundColor: "#FFFFFF",
 
-    "& fieldset": {
-      borderColor: "var(--input-default-color)",
-    },
     "&.Mui-focused fieldset": {
-      borderColor: "#3548d4",
+      borderColor: `${TEXT_FIELD_COLOR_MAP[props.color] || TEXT_FIELD_COLOR_MAP.focused}`,
       borderWidth: "var(--input-border-width)",
     },
+    "&:not(.Mui-focused) fieldset": {
```

Comment:
Looks like this is breaking the hover border color change when there isn't a color selected, you can see the difference here: https://storybook.clutch.sh/?path=/story/core-input-textfield--primary